agents/official/beeai-framework/src/supervisor.ts

```diff
@@ -0,0 +1,143 @@
+import { DynamicTool, JSONToolOutput } from "beeai-framework/tools/base";
+import { BeeAgent } from "beeai-framework/agents/bee/agent";
+import { TokenMemory } from "beeai-framework/memory/tokenMemory";
+
+import {
+  promptInputSchema,
+  promptOutputSchema,
+} from "@i-am-bee/beeai-sdk/schemas/prompt";
+import { z } from "zod";
+import { SSEClientTransport } from "@i-am-bee/acp-sdk/client/sse.js";
+import { Client as MCPClient } from "@i-am-bee/acp-sdk/client/index.js";
+import { OpenAIChatModel } from "beeai-framework/adapters/openai/backend/chat";
+
+const platformSdk = (client: MCPClient, availableAgents: string[]) => {
+  const listAgents = async () => {
+    const agents = await client.listAgents();
+
+    return agents.agents
+      .filter((agent) => availableAgents.includes(agent.name))
+      .map((agent) => ({
+        id: agent.name,
+        description: agent.description,
+      }));
+  };
+
+  const runAgent = async (agent: string, prompt: string) => {
+    console.log(`Called agent - ${agent} and prompting it with - ${prompt}`);
+
+    const agents = await listAgents();
+    if (!agents.map(({ id }) => id).includes(agent)) {
+      throw new Error(`Agent ${agent} is not registered in the platform`);
+    }
+
+    return client.runAgent(
+      { name: agent, input: { prompt } },
+      { timeout: 10000000 }
```

Comment:
We'll need to figure this out.

The problem is that running agents can take a while especially if these are huge as this.

Reply:
Use signal